src/test/java/atdd/path/application/UserServiceTest.java

```diff
@@ -1,55 +1,131 @@
 import atdd.path.domain.User;
+import atdd.path.repository.UserRepository;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.mock.web.MockHttpServletRequest;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.BDDMockito.given;
-import static org.mockito.Mockito.when;
 
-@SpringBootTest(classes = UserService.class)
+@SpringBootTest(classes = {UserService.class, JwtTokenProvider.class})
 public class UserServiceTest {
 
     private UserService userService;
 
     @MockBean
-    private UserDao userDao;
+    private UserRepository userRepository;
+
+    @Autowired
+    private JwtTokenProvider jwtTokenProvider;
 
     @BeforeEach
     void setUp() {
-        this.userService = new UserService(userDao);
+        this.userService = new UserService(userRepository, jwtTokenProvider);
     }
 
     @DisplayName("회원 등록이 된다")
     @Test
     void createUserTest() {
         // given
         UserRequestView userInfo = UserRequestView.builder()
-                .email("boorwonie@email.com")
-                .name("브라운")
-                .password("subway")
+                .email(TestConstant.EMAIL_BROWN)
+                .name(TestConstant.NAME_BROWN)
+                .password(TestConstant.PASSWORD_BROWN)
                 .build();
-        given(userDao.save(any(User.class)))
-                .willReturn(User.createBuilder().email("boorwonie@email.com").name("브라운").build());
+        given(userRepository.save(any(User.class)))
+                .willReturn(User.createBuilder().email(TestConstant.EMAIL_BROWN).name(TestConstant.NAME_BROWN).build());
 
         // when
         UserResponseView response = userService.createUser(userInfo);
 
         // then
-        assertThat(response.getName()).isEqualTo("브라운");
+        assertThat(response.getName()).isEqualTo(TestConstant.NAME_BROWN);
     }
 
     @DisplayName("회원 탈퇴가 된다")
     @Test
     void deleteUserTest() {
+        // give
+        UserRequestView userInfo = UserRequestView.builder()
+                .email(TestConstant.EMAIL_BROWN)
+                .name(TestConstant.NAME_BROWN)
+                .password(TestConstant.PASSWORD_BROWN)
+                .build();
+        given(userRepository.save(any(User.class)))
+                .willReturn(User.createBuilder().id(1L).email(TestConstant.EMAIL_BROWN).name(TestConstant.NAME_BROWN).build());
+        UserResponseView response = userService.createUser(userInfo);
+
+        // when
+        userService.deleteUser(response.getId());
+
+        // then
+        assertThat(userRepository.findById(response.getId()).orElse(null)).isNull();
```

Comment:
userService.deleteUser 내부에서 사용되지 않는 메서드를 호출하면서 검증하려는 것은 userService.deleteUser와 더불어 userRepository.deleteById의 기능까지 확인을 하려함으로 보입니다. 즉, 통합 테스트를 수행하고 계십니다.

검증 방법은 두 가지가 있다고 안내해드렸었습니다.
하나는 사용해주신 상태 검증이고요, 다른 하나는 행위 검증입니다.

userService.deleteUser 메서드의 현재 역할은 userRepository.deleteById을 수행시키는 것이 전부입니다.
따라서 userRepository.deleteById 메서드가 정상적으로 호출되었는지만 확인해보면 되기 때문에 해당 메서드의 행위 검증을 해주면 어떨까요?

이런 경우 반드시 행위 검증을 해야한다는 의미 보다는 행위 검증으로 테스트를 할 수 있다는 메시지를 드리고 싶어 코멘트 남깁니다 :)

```suggestion
        verify(userRepository).deleteById(any());
```